src/Parser.php

```diff
@@ -31,8 +25,8 @@ class Parser
      * @param string|string[] $endSep
      */
     public function __construct(
-        ?YAMLParser $yamlParser = null,
-        ?MarkdownParser $markdownParser = null,
+        YAMLParser|null $yamlParser = null,
+        MarkdownParser|null $markdownParser = null,
```

Comment:
Isn't `?` the same as `|null`?

Reply:
It is, yes, in versions of PHP prior to 8.4. But [PHP 8.4 deprecates implicitly nullable types](https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated).

Reply:
Right, but implicit nullable is `string $test = null`.

What I'm saying is that `?string $test = null` is valid, `?string` being the same as `string|null`. So no change is needed here?

Reply:
Well, as I understand it, yes and no. For PHP versions from 8.3 backwards, yes, it's fine. But, a deprecation notice will be emitted in PHP 8.4. 

And, speaking personally, I agree with https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated#change that using a union type makes the code more verbose and easier to intuit. But, I may be in the minority with this opinion.